.github/workflows/labels.yml

```diff
@@ -128,102 +123,68 @@ jobs:
+                log('URL', item.html_url)
+
+                const pull_number = item.number
+                const issue_number = item.number
+
+                // The search result is of a format that works for both issues and pull requests and thus
+                // does not have all fields of a full pull_request response. Notably, it is missing `head.sha`,
+                // which we need to fetch the workflow run below. When triggered via pull_request event,
+                // this field is already available.
+                // This API request is also important for the merge-conflict label, because it triggers the
+                // creation of a new test merge commit. This is needed to actually determine the state of a PR.
+                const pull_request = item.head ? item : (await github.rest.pulls.get({
+                  ...context.repo,
+                  pull_number
+                })).data
+
+                const run_id = (await github.rest.actions.listWorkflowRuns({
+                  ...context.repo,
+                  workflow_id: 'pr.yml',
+                  event: 'pull_request_target',
+                  status: 'success',
+                  exclude_pull_requests: true,
+                  head_sha: pull_request.head.sha
+                })).data.workflow_runs[0]?.id ??
+                  // TODO: Remove this after 2025-09-17, at which point all eval.yml artifacts will have expired.
+                  (await github.rest.actions.listWorkflowRuns({
                     ...context.repo,
-                    workflow_id: 'pr.yml',
+                    // In older PRs, we need eval.yml instead of pr.yml.
+                    workflow_id: 'eval.yml',
                     event: 'pull_request_target',
-                    // For PR events, the workflow run is still in progress with this job itself.
-                    status: prEventCondition ? 'in_progress' : 'success',
+                    status: 'success',
                     exclude_pull_requests: true,
                     head_sha: pull_request.head.sha
-                  })).data.workflow_runs[0]?.id ??
-                    // TODO: Remove this after 2025-09-17, at which point all eval.yml artifacts will have expired.
-                    (await github.rest.actions.listWorkflowRuns({
-                      ...context.repo,
-                      // In older PRs, we need eval.yml instead of pr.yml.
-                      workflow_id: 'eval.yml',
-                      event: 'pull_request_target',
-                      status: 'success',
-                      exclude_pull_requests: true,
-                      head_sha: pull_request.head.sha
-                    })).data.workflow_runs[0]?.id
-
-                  // Newer PRs might not have run Eval to completion, yet. We can skip them, because this
-                  // job will be run as part of that Eval run anyway.
-                  if (log('Last eval run', run_id ?? '<pending>', !run_id))
```

Comment:
> Wait but #419600 still has no rebuild labels?

[Logs](https://github.com/NixOS/nixpkgs/actions/runs/15854570902/job/44696959165?pr=419600#step:4:227):
```
cutoff timestamp: 1970-01-01T00:00:00.000Z
PR #419600 - Last updated at: 2025-06-24T15:22:45Z
PR #419600 - URL: https://github.com/NixOS/nixpkgs/pull/419600
PR #419600 - Last eval run: <pending> (skipped)
```

Could it be an issue with running just before/after this was merged/reverted? "Last evl run: \<pending\>" is only printed by the current code, not the code added in this PR.